core-bundle/src/File/MetaData.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\File;
+
+/**
+ * @psalm-immutable
+ *
+ * This class is as a container for file meta data as typically defined in
+ * tl_files / tl_content. Its underlying data structure is a key-value store
+ * with added getters/setters for convenience.
+ *
+ * The data must be stored in a normalized form. It's your responsibility to
+ * ensure this is the case when creating an instance of this class. You can
+ * use the public class constants as keys for a better DX.
+ */
+class MetaData
+{
+    public const VALUE_ALT = 'alt';
+    public const VALUE_CAPTION = 'caption';
+    public const VALUE_TITLE = 'title';
+    public const VALUE_URL = 'link';
+
+    /**
+     * Key-value pairs of meta data.
+     *
+     * @var array<string, mixed>
+     */
+    private $values;
+
+    public function __construct(array $values)
+    {
+        $this->values = $values;
+    }
+
+    /**
+     * Get a new meta data container that is the result of merging this
+     * container's data with the data of the specified one.
+     */
+    public function withOther(self $metaData): self
```

Comment:
this does not seem to be used anywhere? I'd refrain from adding "features" that _might be useful at some point_.

Reply:
Agreed. This was a leftover from an earlier version which can be dropped.